### DISCUSSION TITLE
iOS Lesson 4 Segued Screen only shows details of first cell

### DISCUSSION THREAD

**Discussion by ndgsghdj** [sst-inc/help#30] [Q&A - iOS] [answered]
When clicking a cell in the table view controller, the segued screen would only show the details of the first cell, despite the clicked cell being the second or third cell.

https://user-images.githubusercontent.com/75611248/128792782-9cf5bfdc-f230-4c58-920c-9741f213649b.mov
I had expected the segued screen to display the details of the corresponding cell clicked, however, the screen would only display the details of the first cell.
This is my source code:
[Contacts List 2 3.zip](https://github.com/sst-inc/help/files/6958793/Contacts.List.2.3.zip)

**Comment by granwyntan** [comment, marked as answer]
So in your **TableViewController.swift**, have a variable to store the index, then in the `didSelectRowAt() function`, modify the index to be `indexPath`.
<img width="906" alt="Screenshot 2021-08-10 at 11 43 06" src="https://user-images.githubusercontent.com/61010627/128805104-80d82bb7-ce58-4e48-a059-be5504ab5716.png">
Then, you'll need to modify the index accordingly in your `prepare(for segue:, sender:)` override function. 
<img width="906" alt="Screenshot 2021-08-10 at 11 43 22" src="https://user-images.githubusercontent.com/61010627/128805099-45993696-65c7-4875-b7b8-3c3ecff497bf.png">
update the text below `// Send data here` to modify the `index` variable in your **DetailViewController.swift**, which is now with the value of 0

Don't worry if you're still confused you can listen in next lesson as we'll be going through.

**Reply by granwyntan** [reply]
Please mark as answer if this helps
